tests/unit/test_resolver.py

```diff
@@ -93,3 +97,72 @@ def fetch(req: Request) -> RequestOutput:
         assert resolver.resolve_packages(request) == COMBINED_OUTPUT
 
     assert calls_by_pkgtype == ["gomod", "npm", "pip"]
+
+
+@pytest.mark.parametrize(
+    "flags",
+    [
+        pytest.param(["dev-mode"], id="dev-mode-true"),
+        pytest.param([], id="dev-mode-false"),
+    ],
+)
+def test_dev_mode(flags: list[str]) -> None:
+    mock_resolver = mock.Mock()
+    mock_resolver.return_value = RequestOutput.empty()
+    with (
+        mock.patch.dict(
+            resolver._package_managers,
+            values={"gomod": mock_resolver},
+            clear=True,
+        ),
+        mock.patch.dict(
+            resolver._dev_package_managers,
+            values={"shrubbery": mock_resolver},
+            clear=True,
+        ),
+    ):
+        dev_package_input = mock.Mock()
+        dev_package_input.type = "shrubbery"
+
+        request = mock.Mock()
+        request.flags = flags
+        request.packages = [dev_package_input]
+
+        if flags:
+            assert resolver.resolve_packages(request) == RequestOutput(
+                components=[], build_config=BuildConfig(environment_variables=[], project_files=[])
+            )
+        else:
+            expected_error = re.escape("Package manager(s) not yet supported: shrubbery")
+            with pytest.raises(UnsupportedFeature, match=expected_error):
+                resolver.resolve_packages(request)
+
+
+def test_dev_mode_with_released() -> None:
```

Comment:
Nitpick: Rename this test to `test_resolve_with_released_and_dev_package_managers`. More verbose, but I'll probably forget what `released` is without more context :laughing: 

(or maybe just merge this scenario with the previous test... not sure if it'll get too complex)

Reply:
> I'll probably forget what released is without more context

Thoroughly valid :+1: